src/main/java/org/vanilladb/core/util/Timer.java

```diff
@@ -76,7 +91,23 @@ public void reset() {
 		subTimers.clear();
 		componenents.clear();
 	}
+	
+	public void setStartComponentTimer(Object component,long start) {

```

Comment:
Please change the signature to `public void startComponentTimer(Object component, long startTime)` and put this method right after `public void startComponentTimer(Object component)`.